src/libraries/System.Private.CoreLib/src/System/Buffers/Text/Utf8Formatter/Utf8Formatter.Integer.Unsigned.cs

```diff
@@ -25,8 +25,12 @@ private static bool TryFormatUInt64(ulong value, Span<byte> destination, out int
             {
                 case 'G':
                 case 'g':
+                case 'R':
+                case 'r': // treat 'R'/'r' (which historically was only for floating-point) as being equivalent to 'g', rather than throwing
                     if (format.HasPrecision)
+                    {
                         throw new NotSupportedException(SR.Argument_GWithPrecisionNotSupported); // With a precision, 'G' can produce exponential format, even for integers.
```

Comment:
Hmmm. Preventing this seems wrong and inconsistent with what `UTF16` formatting does. (`5.ToString("G99")` is valid)

The official docs (https://docs.microsoft.com/en-us/dotnet/standard/base-types/standard-numeric-format-strings) cover that it produces `Result: The more compact of either fixed-point or scientific notation` and `Precision specifier: Number of significant digits.`

There are notably some other inconsistencies here as well. Integers are missing support for `C`, `E`, `F`, and `P`. `Floating-point` more broadly doesn't support `D` either, but reasonably could (its basically just an alternative to `F0`)

Reply:
-- Noting this is existing and I don't think this PR needs to address it, just calling it out as something else that might need broader consideration.

I didn't realize the UTF8 formatter was "so limited" compared to the UTF16 one.

Reply:
Thanks.  Mind opening a separate issue to track that then?  It sounds like it's something we should address (at some point).

Reply:
https://github.com/dotnet/runtime/issues/67962